### DISCUSSION TITLE
Make service-configuration optional.

### DISCUSSION THREAD

**Discussion by paulincai** [meteor/meteor#12171] [Feature Requests]
It looks that ```accounts-base``` depends on ```service-configuration```. It made sense many years back. It still make sense for many (or few).
However, with the new tech markets, Meteor is used more and more for Pro softwares like POS, business apps, crypto platforms/interfaces etc. In these systems it is either not desired/required or not accepted by businesses to login with third parties. Security has moved more into Passwordless, away from big tech and today you may expect to login with FB only in ... maybe Wordpress ...

While 3rd party login is still imperious for many, the ones that don't need it are penalized with having to install the ```service-configuration ``` package because it is required by ```accounts-base```.

Please untie the two packages and only suggest ```service-configuration``` if you use a 3rd party login.

![Screen Shot 2022-09-01 at 5 10 14 PM](https://user-images.githubusercontent.com/26188915/187921866-12e8f63d-068b-493a-9bbe-a01d8cde3f09.png)

**Comment by StorytellerCZ** [comment]
Agreed. I have taken an initial look and I think most of it could be moved to the oauth package. Though there are one or two places that are not as simple.

**Comment by zodern** [comment]
This is slightly off topic, but it should not be possible for the error in the screenshot to happen and it is a bug. Do you have a reproduction?

**Reply by StorytellerCZ** [reply]
I think I have the location of the error. See #12202
